.github/workflows/ci-tls.yml

```diff
@@ -0,0 +1,111 @@
+        uses: actions/checkout@v3
+        with:
+          path: ./smithy-rs
+      - name: Checkout trytls
+        uses: actions/checkout@v3
+        with:
+          repository: ouspg/trytls
+          path: ./trytls
+      - name: Checkout badtls
+        uses: actions/checkout@v3
+        with:
+          repository: wbond/badtls.io
+          path: ./badtls.io
+      - name: Checkout badssl
+        uses: actions/checkout@v3
+        with:
+          repository: chromium/badssl.com
+          path: ./badssl.com
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.rust_version }}
+      - name: Build badssl.com
+        shell: bash
+        working-directory: badssl.com
+        env:
+          DOCKER_BUILDKIT: 1
+        run: |
+          perl -p -i -e 's/ruby2\.4/ruby2.6/' Dockerfile
+          grep -q 'start of badssl\.test hosts' /etc/hosts || make list-hosts | sudo tee -a /etc/hosts
+          make serve || true
+          ls certs/sets/test/gen/crt/ca-root.crt
+      - name: Build trytls
+        shell: bash
+        working-directory: trytls
+        run: |
+          perl -p -i -e 's!\./runners!runners!' setup.py
+          sed -i '/import platform/a import distro' runners/trytls/utils.py
+          sed -i 's/platform.linux_distribution()/distro.name(), distro.version(), distro.id()/' runners/trytls/utils.py
+          pip3 install -e .
+      - name: Build badtls.io
+        working-directory: badtls.io
+        shell: bash
+        run: |
+          python3 scripts/generate.py badtls.test
+          sudo mkdir /etc/nginx/tls || true
+          sudo mkdir /var/www || true
+          sudo python3 scripts/install.py /etc/nginx/conf.d /etc/nginx/tls /var/www
+          timeout 2m bash ./scripts/local.sh || true
+          echo '#### start of badtls.test hosts ####' | sudo tee -a /etc/hosts
+          echo '127.0.0.1 domain-match.badtls.test wildcard-match.badtls.test san-match.badtls.test dh1024.badtls.test expired-1963.badtls.test future.badtls.test domain-mismatch.badtls.test san-mismatch.badtls.test bad-key-usage.badtls.test expired.badtls.test wildcard.mismatch.badtls.test rc4.badtls.test weak-sig.badtls.test rc4-md5.badtls.test' | sudo tee -a /etc/hosts
+          echo '#### end of badtls.test hosts ####' | sudo tee -a /etc/hosts
+      - name: Create TLS stub
+        shell: bash
+        run: |
+          cargo new stub
+          cd stub
+          cargo add rustls
+          cargo add aws-config
+          cargo add aws-sdk-sts
+          cargo add tokio -F full
+          cargo add rustls-pemfile
+          cargo add exitcode
+          cargo add rustls-native-certs
+          cargo add x509-parser
+          cargo add hyper-rustls -F 'rustls-native-certs,http2'
+          cargo add aws-credential-types -F hardcoded-credentials
+          cargo add aws-smithy-client -F rustls
+          cp ../smithy-rs/tools/ci-resources/tls-stub.rs src/main.rs
```

Comment:
any reason to not check this in and create an actual crate? You might need the `workspace = ["."]` to avoid interfering in other workspaces.

Reply:
It's for always having the updated dependencies automatically. I can create it in the repository though